packages/flutter_test/lib/src/sheet_display.dart

```diff
@@ -0,0 +1,153 @@
+/// 
+///  * [WidgetTester.pumpFrames], which is often used together to display the
+///    frame-by-frame animation of a widget in a test.
+class SheetDisplay extends StatelessWidget {
+  /// Create an [SheetDisplay] using a fixed `target` widget.
+  /// 
+  /// The `cellSize` and `frameIndex` are required.
+  const SheetDisplay({
+    Key key,
+    this.target, 
+    @required this.cellSize,
+    @required this.cellNum,
+  }) : assert(cellSize != null),
+       assert(cellNum != null),
+       super(key: key);
+
+  /// The widget to show animation of.
+  final Widget target;
+
+  /// The size of a cell. Should contain a widget.
+  /// 
+  /// Should not change throughout the test.
+  final Size cellSize;
+
+  /// The number of clones to be displayed.
+  final int cellNum;
+
+  @override
+  Widget build(BuildContext context) {
+    return RepaintBoundary(
+      child: _CellGrid(
+        cellSize: cellSize,
+        children: List<Widget>.generate(cellNum + 1, (int index) {
+          final int i = cellNum - index;
+          return Container(
+            key: ValueKey<int>(i),
+            child: target,
+          );
+        }),
+      )
+    );
+  }
+}
+
+// A grid of fixed-sized cells that are positioned from top left, horizontal-first,
+// until the the entire grid is filled, discarding the remaining children.
+class _CellGrid extends MultiChildRenderObjectWidget {
+  _CellGrid({
+    Key key,
+    @required this.cellSize,
+    List<Widget> children = const <Widget>[],
+  }) : assert(cellSize != null),
+       super(key: key, children: children);
+
+  final Size cellSize;
+
+  @override
+  _RenderCellGrid createRenderObject(BuildContext context) {
+    return _RenderCellGrid(
+      cellSize: cellSize,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RenderCellGrid renderObject) {
+    renderObject
+      .cellSize = cellSize;
+  }
+}
+
+class _CellGridParentData extends ContainerBoxParentData<RenderBox>{
+}
+
+class _RenderCellGrid extends RenderBox with ContainerRenderObjectMixin<RenderBox, _CellGridParentData>,
+                                             RenderBoxContainerDefaultsMixin<RenderBox, _CellGridParentData> {
+  _RenderCellGrid({
+    List<RenderBox> children,
+    Size cellSize,
+  }) : _cellSize = cellSize {
+    addAll(children);
+  }
+
+  Size get cellSize => _cellSize;
+  Size _cellSize;
+  set cellSize(Size value) {
+    _cellSize = value;
+    markNeedsLayout();
+  }
+
+  int get _columnNum {
+    assert(size != null);
+    return (size.width / cellSize.width).floor();
+  }
+
+  int get _rowNum {
+    assert(size != null);
+    return (size.height / cellSize.height).floor();
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! _CellGridParentData)
+      child.parentData = _CellGridParentData();
+  }
+
+  @override
+  void performResize() {
+    size = constraints.biggest;
+    final int maxChildrenNum = _columnNum * _rowNum;
```

Comment:
basically line 123 should be the body of performResize and the rest should be in performLayout